packages/nodes-base/credentials/WhatsAppOAuth2Api.credentials.ts

```diff
@@ -0,0 +1,70 @@
+import type { ICredentialTestRequest, ICredentialType, INodeProperties } from 'n8n-workflow';
+
+export class WhatsAppOAuth2Api implements ICredentialType {
+	name = 'whatsAppOAuth2Api';
+
+	extends = ['oAuth2Api'];
```

Comment:
correct me if I wrong, but oauth2 mechanism is not used, we are using client/app id and secret and manually requesting access token by `appAccessTokenRead`, this should not extend oAuth2Api and just have 2 parameters App ID and App Secret

Reply:
That is correct, I tried getting OAuth2 to work, but wasn't able to.
So we should be able to simplify it.

Reply:
When I changed it to a simplified cred version it stopped working so discarded the changes. Will keep it as is.

Reply:
could you add more details about issue?